src/org/labkey/test/pages/core/login/AuthDialogBase.java

```diff
@@ -0,0 +1,107 @@
+package org.labkey.test.pages.core.login;
+
+import org.labkey.test.Locator;
+import org.labkey.test.components.bootstrap.ModalDialog;
+import org.labkey.test.components.html.Input;
+import org.labkey.test.components.html.ToggleButton;
+import org.labkey.test.params.login.AuthenticationProvider;
+import org.openqa.selenium.WebDriver;
+
+public abstract class AuthDialogBase<T extends AuthDialogBase> extends ModalDialog
+{
+    private final LoginConfigRow _row;
+
+    protected AuthDialogBase(AuthenticationProvider provider, WebDriver driver)
+    {
+        super(getFinder("Configure New " + provider.getProviderName() + " Authentication", driver));
+        _row = null;
+    }
+
+    protected AuthDialogBase(LoginConfigRow row)
+    {
+        super(getFinder("Configure " + row.getDescription(), row.getDriver()));
+        _row = row;
+    }
+
+    private static ModalDialogFinder getFinder(String title, WebDriver driver)
+    {
+        return new ModalDialogFinder(driver).withTitle(title);
+    }
+
+    protected LoginConfigRow getRow()
+    {
+        return _row;
+    }
+
+    public T setDescription(String description)
+    {
+        elementCache().descriptionInput.set(description);
+        return getThis();
+    }
+
+    public String getDescription()
+    {
+        return elementCache().descriptionInput.get();
+    }
+
+    public T setEnabled(boolean enabled)
+    {
+        elementCache().enableToggle.set(enabled);
+        return getThis();
+    }
+
+    public boolean isEnabled()
+    {
+        return elementCache().enableToggle.get();
+    }
+
+    public T clickButtonExpectingError(String buttonText)
+    {
+        Locators.dismissButton(buttonText).waitForElement(this, 2000).click();
+        return getThis();
```

Comment:
`*ExpectingError` methods should wait for and return the error.

Reply:
At this point, that's going to be a trick.
In the SamlConfigureDialog at least, some of the errors are at the field-level, some are text somewhere in the bottom of the dialog.  Not all of the errors are in a predictable or consistent place.

Reply:
Oh, that makes sense.